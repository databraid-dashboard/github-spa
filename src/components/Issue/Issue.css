.space-issues {
  position: relative;
  padding-top: 10px !important;
  padding-bottom: 0px !important;
}

.space-labels {
  position: relative;
  padding-top: 4px !important;
  padding-bottom: 0px !important;
}
