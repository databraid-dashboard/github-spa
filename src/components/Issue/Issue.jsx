import React from 'react';
import { Card, Item, List, Label, Header } from 'semantic-ui-react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import injectWidgetId from '../../utils/utils';
import './Issue.css';

const photo = {
  height: '70px',
  width: '70px',
};

function getDescription(assignedTo) {
  if (assignedTo[0] !== 'Not assigned') {
    return (
      <div className="ui center aligned">
        <div>
          <Header sub>Assigned To: </Header>
          {assignedTo[0]}
        </div>
        <Item.Image style={photo} size="small" src={assignedTo[1]} />
      </div>
    );
  }
  return <div />;
}

export const Issue = ({ title, number, assignedTo, labels }) => {
  const styles = {
    borderTop: '0.25px solid black',
  };

  const readable = {
    fontSize: '16px',
  };

  const padRead = {
    fontSize: '16px',
    paddingTop: '10px',
  };

  const renderedLabels = labels.map(label => (
    <div key={label}>
      <Label className="ui medium label" ribbon="right" color="grey">
        {label}
      </Label>
    </div>));

  return (
    <Card.Content style={styles}>
      <List divided relaxed>
        <List.Item>
          <List.Content>
            <List.Header style={readable}>
            #{number} {title}
            </List.Header>
            {renderedLabels}
            <List.Description style={padRead}>
              {' '}{getDescription(assignedTo)}
            </List.Description>
          </List.Content>
        </List.Item>
      </List>
    </Card.Content>);
};

Issue.propTypes = {
  title: PropTypes.string.isRequired,
  number: PropTypes.number.isRequired,
  assignedTo: PropTypes.arrayOf(PropTypes.string).isRequired,
  labels: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export const mapStateToProps = (state, ownProps) => {
  const id = ownProps.widgetId;
  const issueId = ownProps.issueId;
  const issue = state.widgets.byId[id].issues.issuesById[issueId];
  const { title, number, assignedTo, labels } = issue;

  return {
    title,
    number,
    assignedTo,
    labels,
  };
};

export default injectWidgetId(connect(mapStateToProps)(Issue));
