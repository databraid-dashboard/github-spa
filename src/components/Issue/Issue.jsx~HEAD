import React from 'react';
import { Card, Item, List, Label, Header } from 'semantic-ui-react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './Issue.css';

function getDescription(assignedTo) {
  if (assignedTo.length > 0) {
    return (
      <div className="ui center aligned">
        <div><Header sub className="space-issues">Assigned To: </Header>{assignedTo[0][0]}</div>
        <Item.Image size="mini" src={assignedTo[0][1]} />
      </div>
    );
  }
  return (
    <div />
  );
}

export const Issue = ({ title, repoIssueNumber, assignedTo, labels }) => {
  const renderedLabels = labels.map(label => (
    <div className="space-labels" key={label}>
      <Label className="ui mini label" ribbon="right" color="grey">{label}
      </Label>
    </div>
  ),
  );

  return (
    <Card.Content>
      <List divided relaxed>
        <List.Item>
          <List.Content>
            <List.Header>#{repoIssueNumber} {title}</List.Header>{renderedLabels}
            <List.Description> {getDescription(assignedTo)}</List.Description>
          </List.Content>
        </List.Item>
      </List>
    </Card.Content>
  );
};

Issue.propTypes = {
  title: PropTypes.string.isRequired,
  repoIssueNumber: PropTypes.number.isRequired,
  assignedTo: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string)).isRequired,
  labels: PropTypes.arrayOf(PropTypes.string).isRequired,
};
// NOTE replace Line 43 with below
// assignedTo: PropTypes.arrayOf(PropTypes.number),
export const mapStateToProps = (state, { issueId }) => {
  const issue = state.issues.issuesById[issueId];
  const { title, repoIssueNumber, assignedTo, labels } = issue;
  return {
    title,
    repoIssueNumber,
    assignedTo,
    labels,
  };
};

export default connect(
  mapStateToProps,
)(Issue);
