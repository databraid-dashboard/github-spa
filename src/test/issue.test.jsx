import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { Issue, mapStateToProps } from '../components/Issue/Issue';

const state = {

  issues: {
    issuesById: {
      238923429: {
        assignedTo: [['Gwillison415', 'https://avatars2.githubusercontent.com/u/22106099?v=4']],
        labels: ['duplicate', 'help wanted', 'question'],
        repoIssueNumber: 16,
        title: 'login tst JWT token',

      },
    } },

};

describe('Issue entry component', () => {
  it('should render a component with props as specified ', () => {
    const component = shallow(
      <Issue
        labels={state.issues.issuesById[238923429].labels}
        issueId={238923429}
        assignedTo={[['Gwillison415', 'https://avatars2.githubusercontent.com/u/22106099?v=4']]}
        repoIssueNumber={16}
        title={'login tst JWT token'}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
  it('map\'s given State To Props ', () => {
    const expected = {
      assignedTo: [['Gwillison415', 'https://avatars2.githubusercontent.com/u/22106099?v=4']],
      labels: ['duplicate', 'help wanted', 'question'],
      repoIssueNumber: 16,
      title: 'login tst JWT token',
    };
    expect(mapStateToProps(state, { issueId: 238923429 })).toEqual(expected);
  });

  it('Should have a Cart Header in the center', () => {
    const component = shallow(
      <Issue
        repoIssueNumber={16}
        title={'login tst JWT token'}
        issueId={238923429}
        labels={state.issues.issuesById[238923429].labels}
        assignedTo={[['Gwillison415', 'https://avatars2.githubusercontent.com/u/22106099?v=4']]}
      />,
    );
    expect(component.find('.aligned').exists()).toBe(true);
  });
});
