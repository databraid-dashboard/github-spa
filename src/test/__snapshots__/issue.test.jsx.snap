// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issue entry component should render a component with props as specified  1`] = `
<CardContent>
  <List
    divided={true}
    relaxed={true}
  >
    <ListItem>
      <ListContent>
        <ListHeader>
          #
          16
           
          login tst JWT token
        </ListHeader>
        <div
          className="space-labels"
        >
          <Label
            className="ui mini label"
            color="grey"
            ribbon="right"
          >
            duplicate
          </Label>
        </div>
        <div
          className="space-labels"
        >
          <Label
            className="ui mini label"
            color="grey"
            ribbon="right"
          >
            help wanted
          </Label>
        </div>
        <div
          className="space-labels"
        >
          <Label
            className="ui mini label"
            color="grey"
            ribbon="right"
          >
            question
          </Label>
        </div>
        <ListDescription>
           
          <div
            className="ui center aligned"
          >
            <div>
              <Header
                className="space-issues"
                sub={true}
              >
                Assigned To: 
              </Header>
              Gwillison415
            </div>
            <ItemImage
              size="mini"
              src="https://avatars2.githubusercontent.com/u/22106099?v=4"
            />
          </div>
        </ListDescription>
      </ListContent>
    </ListItem>
  </List>
</CardContent>
`;
